tools/istio-iptables/main.go

```diff
@@ -0,0 +1,516 @@
+	flagSet.StringVar(&inboundPortsExclude, "d", inboundPortsExclude,
+		"Comma separated list of inbound ports to be excluded from redirection to Envoy (optional). "+
+			"Only applies  when all inbound traffic (i.e. \"*\") is being redirected (default to $ISTIO_LOCAL_EXCLUDE_PORTS)")
+	flagSet.StringVar(&outboundIPRangesInclude, "i", outboundIPRangesInclude,
+		"Comma separated list of IP ranges in CIDR form to redirect to envoy (optional). "+
+			"The wildcard character \"*\" can be used to redirect all outbound traffic. An empty list will disable all outbound")
+	flagSet.StringVar(&outboundIPRangesExclude, "x", outboundIPRangesExclude,
+		"Comma separated list of IP ranges in CIDR form to be excluded from redirection. "+
+			"Only applies when all  outbound traffic (i.e. \"*\") is being redirected (default to $ISTIO_SERVICE_EXCLUDE_CIDR)")
+	flagSet.StringVar(&outboundPortsExclude, "o", outboundPortsExclude,
+		"Comma separated list of outbound ports to be excluded from redirection to Envoy (optional")
+	flagSet.StringVar(&kubevirtInterfaces, "k", kubevirtInterfaces,
+		"Comma separated list of virtual interfaces whose inbound traffic (from VM) will be treated as outbound (optional)")
+
+	var dryRun bool
+	flagSet.BoolVar(&dryRun, "dryRun", false, "Do not call any external dependencies like iptables")
+	err := flagSet.Parse(args)
+	if err != nil {
+		return
+	}
+
+	var ext dep.Dependencies
+	if dryRun {
+		ext = &dep.StdoutStubDependencies{}
+	} else {
+		ext = &dep.RealDependencies{}
+	}
+
+	defer func() {
+		ext.RunOrFail(dep.IPTABLESSAVE)
+		ext.RunOrFail(dep.IP6TABLESSAVE)
+	}()
+
+	// TODO: more flexibility - maybe a whitelist of users to be captured for output instead of a blacklist.
+	if proxyUID == "" {
+		usr, err := ext.LookupUser()
+		var userID string
+		// Default to the UID of ENVOY_USER and root
+		if err != nil {
+			userID = "1337"
+		} else {
+			userID = usr.Uid
+		}
+		// If ENVOY_UID is not explicitly defined (as it would be in k8s env), we add root to the list,
+		// for ca agent.
+		proxyUID = userID + ",0"
+	}
+
+	// for TPROXY as its uid and gid are same
+	if proxyGID == "" {
+		proxyGID = proxyUID
+	}
+
+	podIP, err := ext.GetLocalIP()
+	if err != nil {
+		panic(err)
+	}
+	// Check if pod's ip is ipv4 or ipv6, in case of ipv6 set variable
+	// to program ip6tablesOrFail
+	if podIP.To4() == nil {
+		enableInboundIPv6s = podIP
+	}
+	//
+	// Since OUTBOUND_IP_RANGES_EXCLUDE could carry ipv4 and ipv6 ranges
+	// need to split them in different arrays one for ipv4 and one for ipv6
+	// in order to not to fail
+	ipv4RangesExclude, ipv6RangesExclude, err := separateV4V6(outboundIPRangesExclude)
+	if err != nil {
+		panic(err)
+	}
+	if ipv4RangesExclude.IsWildcard {
+		panic("Invalid value for OUTBOUND_IP_RANGES_EXCLUDE")
+	}
+
+	ipv4RangesInclude, ipv6RangesInclude, err := separateV4V6(outboundIPRangesInclude)
+	if err != nil {
+		panic(err)
```

Comment:
is there no better error recovery mechanism besides panic?  Under which error conditions does a panic() occur?

Reply:
See https://github.com/istio/istio/pull/14690#issuecomment-501282427